services/s3/src/main/java/software/amazon/awssdk/services/s3/handlers/PutObjectInterceptor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.handlers;
+
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.interceptor.Context;
+import software.amazon.awssdk.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.interceptor.ExecutionInterceptor;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+/**
+ * Interceptor to add an 'Expect: 100-continue' header to the HTTP Request if it represents a PUT Object request.
+ */
+public class PutObjectInterceptor implements ExecutionInterceptor {
```

Comment:
Can you add a ReviewBeforeRelease. This would be a good idea for any operation with a streaming input so it's probably best handled in the marshaller. This is good until we refactor the marshallers though.

Reply:
+1 added.